catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1722 @@
+                getTestMetacard(),
+                getTestMetacard(),
+                getTestMetacard());
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+
+        List<String> ids = new ArrayList<>();
+        ids.add("someMadeUpId");
+        ids.add("anotherOne");
+        ids.add("oneMore");
+
+        List<Metacard> metacards = fasi.getLocalRegistryMetacardsByRegistryIds(ids);
+
+        assertThat(metacards, hasSize(3));
+
+        verify(catalogFramework).query(any(QueryRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testGetLocalRegistryMetacardsByRegistryIdsWithEmptyList() throws Exception {
+        List<String> ids = new ArrayList<>();
+
+        fasi.getLocalRegistryMetacardsByRegistryIds(ids);
+
+        verify(catalogFramework, never()).query(any(QueryRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testGetLocalRegistryMetacardsByRegistryIdsWithSourceUnavailableException()
+            throws Exception {
+        List<String> ids = new ArrayList<>();
+        ids.add("someMadeUpId");
+        ids.add("anotherOne");
+        ids.add("oneMore");
+
+        when(catalogFramework.query(any(QueryRequest.class))).thenThrow(SourceUnavailableException.class);
+
+        fasi.getLocalRegistryMetacardsByRegistryIds(ids);
+
+        verify(catalogFramework).query(any(QueryRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testGetLocalRegistryMetacardsByRegistryIdsWithUnsupportedQueryException()
+            throws Exception {
+        List<String> ids = new ArrayList<>();
+        ids.add("someMadeUpId");
+        ids.add("anotherOne");
+        ids.add("oneMore");
+
+        when(catalogFramework.query(any(QueryRequest.class))).thenThrow(UnsupportedQueryException.class);
+
+        fasi.getLocalRegistryMetacardsByRegistryIds(ids);
+
+        verify(catalogFramework).query(any(QueryRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testGetLocalRegistryMetacardsByRegistryIdsWithFederationException()
+            throws Exception {
+        List<String> ids = new ArrayList<>();
+        ids.add("someMadeUpId");
+        ids.add("anotherOne");
+        ids.add("oneMore");
+
+        when(catalogFramework.query(any(QueryRequest.class))).thenThrow(FederationException.class);
+
+        fasi.getLocalRegistryMetacardsByRegistryIds(ids);
+
+        verify(catalogFramework).query(any(QueryRequest.class));
+    }
+
+    @Test
+    public void testGetLocalRegistryObjects() throws Exception {
+        Metacard localMetacardOne = getTestMetacard();
+        localMetacardOne.setAttribute(new AttributeImpl(Metacard.METADATA, "xmlString"));
+
+        Metacard localMetacardTwo = getTestMetacard();
+        localMetacardTwo.setAttribute(new AttributeImpl(Metacard.MET
```

Comment:
Another slight nitpick, but not too fond of the names of the variables in the tests. For example, this one is called `firstMetacard` but there's never another metacard in this test. Not a deal breaker, but makes reading it a bit strange. :)